weeks/week03/regular_expressions.md

```diff
@@ -0,0 +1,183 @@
+# Section 2: Regular Expressions
+
+Regular expressions (regex) are sequences of characters that define search patterns. They are used for string matching and manipulation. Python provides the `re` module to work with regular expressions.
+
+## References:
+
+1. [Regular Expressions: Regexes in Python - Part 1](https://realpython.com/regex-python/)
+2. [Regular Expressions: Regexes in Python - Part 2](https://realpython.com/regex-python-part-2/)
+3. [re — Regular expression operations - Python Documentation](https://docs.python.org/3/library/re.html)
+4. [Regular Expression HOWTO - Python Documentation](https://docs.python.org/3/howto/regex.html#regex-howto)
+
+---
+
+Regular expressions are powerful tools for matching, searching, and manipulating text. They are widely used in data validation, parsing, and text processing. Now suppose you need to write Python code to find out whether a log record contains the substring 'HttpStatusCode: xxx' where http code can be any three consecutive decimal digit characters.
+
+```python
+log_record = "some text HttpStatusCode: 202 some text"
+```
+
+## 2.1 Using and Compiling Regular Expressions
+
+To use regular expressions in Python, you need to import the `re` module. The `re.compile()` function is used to compile a regular expression pattern into a regex object.
+
+### Syntax:
+```python
+import re
+pattern = re.compile(pattern, flags=0)
```

Comment:
What about adding 
re.escape(char) - to match special symbols 
At least mention it.